### PR TITLE
Revert "pkg/cvo/updatepayload.go: timeout payload retrieval"

### PR DESCRIPTION
**Pull request by petr-muller**
Reverts openshift/cluster-version-operator#846

Apparently introduced a regression that blocks some upgrades, opening to allow `/payload` testing of `periodic-ci-openshift-release-master-nightly-4.13-e2e-metal-ipi-upgrade-ovn-ipv6`. If it fixes the job we should consider merging the revert and work on the fix with clean main branch.

### CODE REVIEW

**Comment by petr-muller** [comment]
/payload-job periodic-ci-openshift-release-master-nightly-4.13-e2e-metal-ipi-upgrade-ovn-ipv6

**Comment by petr-muller** [comment]
This `/payload` thing is really nifty, the people who built it must be awesome humans!

**Comment by petr-muller** [comment]
Removing the bug reference because this is not an actual fix for the bug and the automation started to act on it in the bugzilla.

File: pkg/cvo/updatepayload.go [hunk 1]
```diff
@@ -89,12 +89,22 @@ func (r *payloadRetriever) RetrievePayload(ctx context.Context, update configv1.
 	if index := strings.LastIndex(update.Image, "@"); index != -1 {
 		releaseDigest = update.Image[index+1:]
 	}
+	verifyCtx := ctx
+
+	// if 'force' specified, ensure call to verify payload signature times out well before parent context
```
Comment (deads2k) [inline pkg/cvo/updatepayload.go:94]: This comment looks exactly like what I blindly suggested to Trevor :)

File: pkg/cvo/updatepayload.go [hunk 2]
```diff
@@ -112,7 +122,7 @@ func (r *payloadRetriever) RetrievePayload(ctx context.Context, update configv1.
 
 	// download the payload to the directory
 	var err error
-	info.Directory, err = r.targetUpdatePayloadDir(retrieveCtx, update)
```
Comment (deads2k) [inline pkg/cvo/updatepayload.go:115]: This is bugged.  It couples the lookup timeout to the payload retrieval timeout, causing guaranteed failures when verifies timeout. This prevents force from working without signatures.  Before reintroducing, you should write a test.

**Comment by petr-muller** [comment]
/override ci/prow/e2e-agnostic-upgrade-into-change 

Failures unrelated to CVO:

```
: [sig-node] nodes should not go unready after being upgraded and go unready only once
: [sig-network-edge] ns/openshift-authentication route/oauth-openshift disruption/ingress-to-oauth-server connection/new should be available throughout the test
: [sig-network-edge] ns/openshift-console route/console disruption/ingress-to-console connection/new should be available throughout the test 
: [sig-network-edge] ns/openshift-console route/console disruption/ingress-to-console connection/reused should be available throughout the test
```

**Comment by evakhoni** [comment]
@petr-muller this is just a revert so no qa on this one right?

**Comment by petr-muller** [comment]
@evakhoni that's correct

**Pull request merged**